src/main/java/tigase/server/xmppsession/SessionManager.java

```diff
@@ -401,6 +403,29 @@ public void handleResourceBind(XMPPResourceConnection conn) {
 				// This actually should not happen... might be a bug:
 				log.log(Level.WARNING, "This should not happen, check it out!, ", ex);
 			}
+
+			checkSingleUserConnectionsLimit(conn);
+		}
+	}
+
+	protected void checkSingleUserConnectionsLimit(XMPPResourceConnection conn) {
+		if (getSingleUserConnectionsLimit() != null) {
+			XMPPSession session = conn.getParentSession();
+			if (session != null) {
+				int overlimit = session.getActiveResourcesSize() - getSingleUserConnectionsLimit();
+				if (overlimit > 0) {
+					session.getActiveResources().stream().sorted(Comparator.comparing(XMPPResourceConnection::getCreationTime)).limit(overlimit).forEach(connToStop -> {
```

Comment:
Considering that `tigase.xmpp.XMPPSession#activeResources` is almost "queue-like" (new resources, i.e. elements with shortest creation time) are always added to the end of the list, do we need to sort it here as well?

Reply:
We could drop sorting from here, but most likely sorting will be very fast on a few rows so that would change almost nothing and current code ensures that collection is sorted.